python/cuml/manifold/umap.pyx

```diff
@@ -706,14 +746,27 @@ class UMAP(Base):
 
         """
         if len(X.shape) != 2:
-            raise ValueError("data should be two dimensional")
+            raise ValueError("X should be two dimensional")
+
+        if is_sparse(X) and not self._sparse_fit:
```

Comment:
maybe we could support this via the csr2dense primitive?

Reply:
I've gone back and forth about this quite a bit. IMO, I feel it's dangerous to automatically convert any sparse data to dense unless there's a batching mechanism handy that the user can control. For now, we can try converting the data automatically in the python layer and printing warnings so the users have a clear picture of what's going on (in case, for example, a sparse->dense conversion blows the memory unexpectedly).